types/chrome-remote-interface/index.d.ts

```diff
@@ -111,7 +197,12 @@ declare namespace CDP {
         Tracing: ProtocolProxyApi.TracingApi;
         WebAudio: ProtocolProxyApi.WebAudioApi;
         WebAuthn: ProtocolProxyApi.WebAuthnApi;
-    }
+    } & EventPromises<ProtocolMappingApi.Events>;
+
+    // '<domain>.<event>' i.e. Page.loadEventFired
+    type EventPromises<T extends ProtocolMappingApi.Events> = {
+        [Property in keyof T]: T[Property] extends any[] ? Promise<T[Property][0]> : never;
```

Comment:
```suggestion
        [Property in keyof T]: T[Property] extends [any] ? Promise<T[Property][0]> : Promise<void>;
```

Reply:
```suggestion
        [Property in keyof T]: T[Property] extends [any, ...any[]] ? Promise<T[Property][0]> : Promise<void>;
```

according to https://stackoverflow.com/questions/49910889/typescript-array-with-minimum-length/67026991#67026991.

Reply:
The minimum length is actually 0, so this change is not needed.
https://github.com/ChromeDevTools/devtools-protocol/blob/master/types/protocol-mapping.d.ts#L27

Reply:
Practically yes, but the type should be `Promise<void>` imo if the length is 0 ... `any[]` matches even `[]`.